tools/clang/unittests/HLSL/OptionsTest.cpp

```diff
@@ -380,3 +384,436 @@ TEST_F(OptionsTest, TestPreprocessOption) {
   VerifyPreprocessOption("/T ps_6_0 -P out.pp input.hlsl", "out.pp", Warning);
   VerifyPreprocessOption("/T ps_6_0 input.hlsl -P out.pp ", "out.pp", Warning);
 }
+
+static void VerifySerializeDxilFalgs(llvm::StringRef command,
+                                     uint32_t ExpectFlags) {
+  std::string errorString;
+  const llvm::opt::OptTable *optionTable = getHlslOptTable();
+  llvm::SmallVector<llvm::StringRef, 4> args;
+  command.split(args, " ", /*MaxSplit*/ -1, /*KeepEmpty*/ false);
+  MainArgs argStrings(args);
+  DxcOpts dxcOpts;
+  llvm::raw_string_ostream errorStream(errorString);
+
+  int retVal =
+      ReadDxcOpts(optionTable, DxcFlags, argStrings, dxcOpts, errorStream);
+  EXPECT_EQ(retVal, 0);
+  errorStream.flush();
+  printf("%s\n", errorString.c_str());
+  EXPECT_TRUE(errorString.empty());
+  EXPECT_EQ(
+      static_cast<uint32_t>(hlsl::options::ComputeSerializeDxilFlags(dxcOpts)),
+      ExpectFlags);
+}
+
+TEST_F(OptionsTest, SerializeDxilFlags) {
+  VerifySerializeDxilFalgs(
+      "-T lib_6_3 input.hlsl -Qstrip_rootsignature -Qkeep_reflect_in_dxil "
+      "-Qstrip_reflect -Zss -Zs",
+      static_cast<uint32_t>(SerializeDxilFlags::StripRootSignature) |
```

Comment:
I'm finding it hard to understand the intent of each of these tests and determining the overall coverage.

One thing that confuses me is the repeated 

```
    static_cast<uint32_t>(SerializeDxilFlags::None) |
    static_cast<uint32_t>(SerializeDxilFlags::None) |
```
in some of the tests.

Is this trying to test all of the combinations? Its not clear to me that is the case and if not why were these specific examples chosen?

Maybe we could use a table driven approach here to make it a bit more concise about what combinations we are testing and why?

```
CombineFlags(ArrayRef<SerializeDxilFlags> flags)
{
   uint32_t combined = 0;
   for (auto f : flags) combined |= static_cast<uint32_t>(f);
  return combined;
}

using F=SerializeDxilFlags;
struct Test {char *flags, uint32_t expected};
Test Tests[] = {
  {"",
    CombineFlags({
        F::StripReflectionFromDxilPart,
        F::IncludeReflectionPart}) },
  {"-Qstrip_rootsignature -Qkeep_reflect_in_dxil -Qstrip_reflect -Zss -Zs",
    CombineFlags({
        F::StripRootSignature,
        F::IncludeDebugNamePart,
        F::DebugNameDependOnSource}) },
}

for (const Test &T : Tests) {
    string cmd = "-T lib_6_3 input.hlsl" + T.flags;
    VerifySerializeDxilFlags(cmd, T.expected)
}

Reply:
Fixed.